src/main/java/frc/robot/subsystems/AlgeaRemover.java

```diff
@@ -0,0 +1,155 @@
+
+  public AlgeaRemover() {
+    algeaMotor = new SparkMax(Constants.CANIDs.kAlgeaMotor, MotorType.kBrushless);
+    algeaMotor.configure(
+        new SparkMaxConfig().apply(algeaCfg),
+        ResetMode.kNoResetSafeParameters,
+        PersistMode.kNoPersistParameters);
+    if (Robot.isSimulation()) {
+      simAlgeaMotor = new SparkMaxSim(algeaMotor, AlgeaRemoverConstants.kAlgeaGearbox);
+      algeaSim =
+          new SingleJointedArmSim(
+              AlgeaRemoverConstants.kAlgeaGearbox,
+              AlgeaRemoverConstants.kAlegeaGearRatio,
+              SingleJointedArmSim.estimateMOI(
+                  AlgeaRemoverConstants.algeaArmLength.in(Meters), .5), // Update with real value
+              AlgeaRemoverConstants.algeaArmLength.in(Meters),
+              -Math.PI / 2,
+              Math.PI / 4,
+              true,
+              -Math.PI / 2);
+      algeaMech =
+          mech.getRoot("root", 1, 1)
+              .append(
+                  new MechanismLigament2d(
+                      "Algea Mech [0]", 1, -90, 10, new Color8Bit(Color.kDarkViolet)));
+    }
+    if (widget == null) {
+      widget = Shuffleboard.getTab(getName()).add("Algea Remover", mech);
+    }
+  }
+
+  public Command goUp() {
+    return startEnd(
+        () -> {
+          System.out.println("up");
+          algeaMotor.set(AlgeaRemoverConstants.kAlgeaPercent);
+        },
+        () -> algeaMotor.set(0));
+  }
+
+  public Command goDown() {
+    return startEnd(
+        () -> {
+          System.out.println("down");
+          algeaMotor.set(-AlgeaRemoverConstants.kAlgeaPercent);
+        },
+        () -> algeaMotor.set(0));
+  }
+
+  public Command zeroAlgeaEncoder() {
+    return Commands.runOnce(
+        () -> {
+          algeaMotor.getAlternateEncoder().setPosition(0);
+        });
+  }
+
+  public Command changeAngle(Angle angle) {
+    return runOnce(
+        () -> {
+          algeaMotor
+              .getClosedLoopController()
+              .setReference(
+                  angle.in(Rotations) * AlgeaRemoverConstants.kAlegeaGearRatio,
+                  ControlType.kPosition);
+
+          SmartDashboard.putNumber(
+              "Algea/Setpoint (Rotations)",
+              angle.in(Rotations) * AlgeaRemoverConstants.kAlegeaGearRatio);
+          SmartDashboard.putNumber(
+              "Algea/Setpoint (Degrees)",
+              angle.in(Degrees) * AlgeaRemoverConstants.kAlegeaGearRatio);
+        });
+  }
+
+  public Command stowAlgea() {
+    return changeAngle(AlgeaRemoverConstants.algeaStowed);
+  }
+
+  public Command removeAlgea() {
+    return changeAngle(AlgeaRemoverConstants.algeaRemove);
+  }
+
+  @Override
+  public void periodic() {
+    SmartDashboard.putNumber("Algea/Motor Rotation", algeaMotor.getEncoder().getPosition());
+    SmartDashboard.putNumber("Algea/Motor Percent", algeaMotor.get());
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    algeaSim.setInputVoltage(algeaMotor.getBusVoltage());
```

Comment:
BusVoltage is the same as the battery voltage. You want to use getAppliedOutput() times the battery voltage to get the applied voltage.
```suggestion
    algeaSim.setInputVoltage(algeaMotor.getBusVoltage() * algeaMotor.getAppliedOutput());
```